### PR TITLE
fix: prevent substring matches in element links

### PR DESCRIPTION
**Pull request by tpatel**
- Sorting by descending length helps avoiding matching substrings, because the longuest strings are searched first

before

Screenshot 2023-06-15 at 10 23 34

after

Screenshot 2023-06-15 at 10 24 02

### CODE REVIEW

**Review by willydouhard — COMMENTED** [review]
Awesome! Can we have a small test for that?

File: src/chainlit/frontend/src/components/chat/message/content.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            type: 'text',
+            content: 'hi'
+          } as ITextElement
+        ]}
+        content={`Hello world
+        source_121, source_1, source_12`}
+      />
+    </BrowserRouter>
+  );
+  expect(getByText('source_1')).toBeInTheDocument();
```
Comment (willydouhard) [inline src/chainlit/frontend/src/components/chat/message/content.spec.tsx:50]: does it check if the text is in the dom or if an element wraps this text specifically (Link in our case)?

**Review by willydouhard — COMMENTED** [review]
Looks good!

File: src/chainlit/frontend/src/components/chat/message/content.spec.tsx (hunk 1 above)
Comment (tpatel) [reply src/chainlit/frontend/src/components/chat/message/content.spec.tsx:50]: `getByText` checks there is only one match with the exact text 👌

**Review by willydouhard — APPROVED** [review]

**Pull request merged**